ChakraCore Samples/JSRT Hosting Samples/C#/ChakraCoreHost/ChakraCoreHost.cs

```diff
@@ -43,6 +74,23 @@ private static JavaScriptValue Echo(JavaScriptValue callee, bool isConstructCall
             return JavaScriptValue.Invalid;
         }
 
+        private static JavaScriptValue SetTimeout(JavaScriptValue callee, bool isConstructCall, JavaScriptValue[] arguments, ushort argumentCount, IntPtr callbackData)
+        {
+            // setTimeout signature is (callback, after)
+            if (arguments.Length < 2)
+            {
+                Console.WriteLine("Invalid argument lengths, required 2");
+                return JavaScriptValue.Invalid;
+            }
+            JavaScriptValue callbackValue = arguments[1];
+            JavaScriptValue timeout = arguments[2];
+            JavaScriptValue afterValue = arguments[2].ConvertToNumber();
+            int after = (int)Math.Max(afterValue.ToDouble(), 1.0);
+            TimeoutCall to = new TimeoutCall(after, callbackValue, callee);
+            timeouts.Add(to.dt,to);
+            return JavaScriptValue.Invalid;
```

Comment:
Please add a note indicating that return value cannot be used as cookie for clearTimeout. 
It would be nice if we can hide the dt as getter/setter.